### ISSUE TITLE
List publishing not generating all frequency versions

### ISSUE THREAD

**Issue by justinlewis** [terraframe/geoprism-registry#186]
**List the CGR version information**
- CGR version: 0.7.0

**Describe the bug (clear and concise)**
See screenshot below

**Screenshots**

![Screenshot from 2020-04-22 13-58-34.png](https://images.zenhubusercontent.com/5df81cb2f5388bdc79385d22/8ac2bb34-7ed7-46ce-bbd5-7ba2e15f079e)

Submitted by @dtreatman

**Comment by dtreatman** [comment]
Note: same issue with publishing lists on other frequencies. This list was set to Quarterly frequency, and as you can see in the screenshot, only the first quarter list was published.

![image.png](https://images.zenhubusercontent.com/5dfcfd403da7a2fba63fee9a/ad5d2aa9-f825-4ae1-823f-ab132cfcf548)

**Comment by SteeveEbener** [comment]
Just a guess: maybe that this is linked to the absence of a starting date that can be set in the past

**Comment by jsmethie** [comment]
The page just needs to be refreshed once the scheduled job is finished to see the new versions.

**Comment by justinlewis** [comment]
@dtreatman See the last comment by smethie.

**Comment by jsmethie** [comment]
Marking as invalid

**Comment by SteeveEbener** [comment]
This is not the case here as the starting date for the data is Jan 1, 2000 => the system should have produced lists for Jan 1, 2001; Jan 1, 2002; Lan 1, 2003;... which is not the case here and the refresh does not make such instances of the list appear

**Comment by justinlewis** [comment]
@steeveebener and I discussed this briefly off of this ticket. This functionality is behaving as developed to prevent empty lists and duplicate lists. If we want to change this functionality we need to create a new ticket to enhance the existing functionality and then have it prioritized into an upcoming sprint.

**Comment by SteeveEbener** [comment]
@justinlewis. There might then be some difference of understanding on what this functionality is meant to do in the first place. This should be discussed

**Comment by dtreatman** [comment]
To clarify expected use that I had understood: if the GeoObject has a
period of one year, I would not expect the user to have to re-upload the
same data for multiple different years in order to populate the system with
data that remains valid across multiple years. If I have a spreadsheet that
has data that is valid for the past 5 years, I would expect to only upload
those data once, with a "For Date" of "Jan 1, 2015", and expect to see
those data for every year between 2015 and today.

@justinlewis, with the current functionality: If I upload data with a "For
Date" of Jan 1, 2020, time passes and we are now in 2021, and I click the
Publish Lists button, what happens? I would expect a new list for 2021 to
pop up, with all the most recent valid data, but it sounds like that
currently won't happen. Am I correct in this assumption?

**Comment by SteeveEbener** [comment]
@dtreatman: This confirms the misunderstanding.

This issue has been part of the COT discussion. 

The frequency information that you enter when creating a list does not relate to the upload of data in the graph but the access to such data ("publishing") by users that do not have the RA/RM/RC role on such data 

While the document has never been finalized, v2 of the COT specification document does describe what is expected in a correct way (page 9, https://docs.google.com/document/d/1aroPzxNLinYKEa242AbDnJdkMbLMXI-AdxJmGaOh8lM/edit?usp=sharing):

"Publishing Master Lists:

- Currently only one master list can be published for a given GeoObject type. With the added support for change over time, master lists can be published for different intervals of frequency specified in the metadata of the master list . For example, a master list with an annual frequency will provide the user with one master list per year. When clicking on a master list in the master list manager, the user will be able to see all of the published lists for that type. within this view,.

- Only a Registry Administrator (RA) or a Registry Maintainer (RM) will be able to look at working version of the master list at any point in time as modeled in the graph database. Each time the RA is confident that the graph has been improved significantly across time he or she generates a new version of the published master lists, one for each of the interval that has been defined based on the frequency attached to master list."

**Comment by justinlewis** [comment]
It was discussed on June 1, 2020 and decided that this ticket should focus on generating lists for every period of the frequency since the last known date where data exists. This could result in duplicate data but will more clearly provide access to lists by frequency period.

**Comment by dtreatman** [comment]
confirmed - discussed with Justin on June 30, we agree with Steeve and Justin's comments above and this is priority from our perspective. As Steeve noted above (note: I added the RC role in as a role that should be able to create an exploratory list, as per the rights by role matrix):

Publishing Master Lists:

Currently only one master list can be published for a given GeoObject type. With the added support for change over time, master lists can be published for different intervals of frequency specified in the metadata of the master list . **For example, a master list with an annual frequency will provide the user with one master list per year.** When clicking on a master list in the master list manager, the user will be able to see all of the published lists for that type. within this view,.

**An RA (for any GeoObject Type for that organization), RM (for org-assigned GeoObject Types), or RC (for org-assigned GeoObject Types)** will be able to look at working version of the master list at any point in time as modeled in the graph database. Each time the RA is confident that the graph has been improved significantly across time he or she generates a new version of the published master lists, **one for each of the interval that has been defined based on the frequency attached to master list**."

**Comment by SteeveEbener** [comment]
This ticket has been implemented except and should be closed.

This being said, we should really implement ticket #265 soon as it does not make sense for the first available list to be dated 12.31.2000 when the validity period of the data starts from 01.01.2000 unless this has been specified as such when creating the list.

**Comment by CarlyNel** [comment]
Closing as per Steeve's request.
